### ISSUE TITLE
FXIOS-1501 ⁃ RustPlaces getBookmarksTree not working in a fork

### ISSUE THREAD

**Issue by p-bertran** [mozilla-mobile/firefox-ios#8002]
Hi, I'm maintaining a fork of this repo.

I'm having trouble with the bookmark panel: nothing appears here (but bookmarks are definitely saved as the add/delete bookmark menu works well).

It appears that it is because places.getBookmarksTree fails returning a BookmarkFolder but with no error.
I'm entering this "else": https://github.com/mozilla-mobile/firefox-ios/blob/173c38770494482b8d60dbd28cbe29462d75ab63/Client/Frontend/Library/BookmarksPanel.swift#L172
I have logged this values, maybe it can help:
```
result.isSuccess = true
result.isFailure = false
result.failureValue = nil
result.description = Result.Success(Optional(MozillaAppServices.BookmarkFolder))
result.successValue.debugDescription = Optional(Optional(MozillaAppServices.BookmarkFolder))
```

I tried calling getBookmarksTree with BookmarkRoots.RootGUID and BookmarkRoots.MobileFolderGUID, with or without recursion, it always does the same.
All bookmarks are saved directly under BookmarkRoots.MobileFolderGUID

I haven't yet gone further as this then calls the [getBookmarksTree](https://github.com/mozilla/application-services/blob/3fc42b212b12493e4113ed787cfa985b0b269a24/components/places/ios/Places/Places.swift#L316) method from the [MozillaAppServices](https://github.com/mozilla/application-services) framework

FxA and sync are disabled in my fork, could it be the reason ?

Any help would be greatly appreciated

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/FXIOS-1501)

**Comment by boek** [comment]
Hey @p-bertran,

This looks like it might be a bug on our side! 

`result.successValue` is containing a double optional for some reason. 
<img width="786" alt="image" src="https://user-images.githubusercontent.com/1250545/108394021-1afbb200-71c9-11eb-886c-b1606fd0e191.png">

You _might_ be able to try something like this in the meantime:

<img width="507" alt="image" src="https://user-images.githubusercontent.com/1250545/108394663-ce64a680-71c9-11eb-8c90-7cf1afebe5a0.png">

and let me know here if that fixes your bug? :)

**Comment by p-bertran** [comment]
I just tried your suggestion, still the same behaviour.

I assume indeed that it is related to the fork somehow, as your app on the store doesn't have the issue.
But the modification done on my side are mostly strings/images and removing FxA and telemetry. I haven't touch any of the bookmarks or rust places files so I'm not sure how I could have impacted this.

I also use the app services provided with Carthage.
Could it be a MozillaAppService version issue ? (I'm based on the [v30](https://github.com/mozilla-mobile/firefox-ios/releases/tag/v30.0) tag of this repo).

**Comment by q2r5** [comment]
I believe this is fixed in main, but not in a release yet. Fixed by #7908.

**Comment by p-bertran** [comment]
It is, with these commits I could fix on my fork too.
Thanks a lot for your time.
